### PR TITLE
Implemented lock delay enforced

### PR DESCRIPTION
**Pull request by fesave**
## Related Issues
App: #3440 

Library PR (if needed):

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

## QA

QA checks: https://github.com/owncloud/android/pull/3547#issuecomment-1044450523

- [x] (1) Option unlocked after enabling passcode https://github.com/owncloud/android/pull/3547#issuecomment-1044485826 [FIXED]

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt:318]: I wouldn't rely on ordinals, let's parse the integer to a LockTimeout and use the advantages of enum classes. We will have an exhaustive `when` if we do it that way. What do you think?

**Review by abelgardep — CHANGES_REQUESTED** [review]
Some suggestions here @fesave

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/SecurityUtils.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/SecurityUtils.kt:32]: Removed

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt:277]: Changed

File: owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt
Comment (fesave) [reply owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt:177]: Removed

File: owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt
Comment (fesave) [reply owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt:173]: Changed

File: owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt
Comment (fesave) [reply owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt:37]: Removed

File: owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModel.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModel.kt:52]: Renamed

File: owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt:309]: Renamed

File: owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt:318]: Changed

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Some requested changes and comments here @fesave! Good job 👍

File: changelog/unreleased/3440
Comment (fesave) [reply changelog/unreleased/3440:3]: I have done the work of my future self, but if you think it is not necessary, I omit it.

File: owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt:311]: _The order of factors does not alter the value of the product,_ Changed.

File: owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt:329]: I did it in this way due to the existing code that I think that you coded in the past:

`val timeout = LockTimeout.valueOf(preferencesProvider.getString(PREFERENCE_LOCK_TIMEOUT, LockTimeout.IMMEDIATELY.name)!!).toMilliseconds()`

I have seen that you retrieve a string value, and this is the reason why I did it.

File: owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModel.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModel.kt:52]: Okay! changed.

File: changelog/unreleased/3440
Comment (JuancaG05) [reply changelog/unreleased/3440:3]: Yes please, we don't know who will develop this and if they will be aware that it is already on changelog, so to avoid repetitions, let's only add what we have done in this PR 👍

File: owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt
Comment (JuancaG05) [reply owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt:311]: Well you know in English it does 😉  `lockDelayEnforced` means you have enforced the lock delay, but `lockEnforcedDelay` could mean that you have enforced the lock and that it does have a delay, which is not the same as enforce the lock delay

File: owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt
Comment (JuancaG05) [reply owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt:329]: Ok, let's keep consistency then 👍

File: owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt
Comment (JuancaG05) [inline owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt:179]: Thanks for changing the name of these 2 tests, but could you change as well the one of the 2 previous tests which are wrongly named too? 👍

File: owncloudApp/src/main/java/com/owncloud/android/MainApp.kt
Comment (JuancaG05) [inline owncloudApp/src/main/java/com/owncloud/android/MainApp.kt:163]: 🏅

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt
Comment (JuancaG05) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt:104]: Great! 🥳

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Last tiny change, great job @fesave! 🏅

File: owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt
Comment (fesave) [reply owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsSecurityViewModelTest.kt:179]: Oh! I forgot them! 😞 , thank you! renamed!

**Review by JuancaG05 — APPROVED** [review]
Great work! Let's wait for @abelgardep's approval and move it on! 💯

**Review by abelgardep — CHANGES_REQUESTED** [review]
Last two suggestions @fesave 👍

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/SecurityUtils.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/SecurityUtils.kt:36]: Yes, exactly. We need a mode that lets the user select an interval mode. This is the only reason to have a `DISABLED` mode

File: owncloudApp/src/main/java/com/owncloud/android/MainApp.kt
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/MainApp.kt:338]: You are absolutely right, cleaner than before, excellent catch. Sometimes we try to make the code more understandable and forget the simplest way. Thank you.

**Review by abelgardep — APPROVED** [review]
Ready to QA -> @jesmrec

**Comment by jesmrec** [comment]
### (1)

1. Set `<integer name="lock_delay_enforced">1</integer>`. (enforced to immediately)
2. install app from scratch
3. Open Settings (passcode and pattern are disabled)
4. Enable pattern or passcode 

Current: 

`Lock Application` is not locked

Expected:

`Lock Application` is locked

Nexus 6P v7, Pixel2 V11
`f19b589`

**Comment by fesave** [comment]
> ### (1)
> 1. Set `<integer name="lock_delay_enforced">1</integer>`. (enforced to immediately)
> 2. install app from scratch
> 3. Open Settings (passcode and pattern are disabled)
> 4. Enable pattern or passcode
> 
> Current:
> 
> `Lock Application` is not locked
> 
> Expected:
> 
> `Lock Application` is locked
> 
> Nexus 6P v7, Pixel2 V11 `f19b589`

Fixed @jesmrec

**Comment by jesmrec** [comment]
Approved on my side. Going on!

**Pull request merged**
